oracle/oracle.go

```diff
@@ -0,0 +1,84 @@
+package oracle
+
+import (
+	"github.com/algorand/go-algorand-sdk/stateproofs/stateproofverification"
+	"github.com/algorand/go-algorand-sdk/stateproofs/transactionverificationtypes"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+// Oracle is in charge of ingesting state proofs, in a chain, and saving their block interval attesting
+// vector commitment roots in a window.
+// It then allows, given a round, to retrieve the vector commitment root attesting to the interval to which the round
+// belongs.
+type Oracle struct {
+	// BlockIntervalCommitmentHistory is a sliding window that holds block interval commitments for each interval. Given a round,
```

Comment:
"BlockIntervalCommitmentHistory is a sliding window of verified block interval commitments. Given a round, it returns the block interval commitment that contains the specified block.